src/main/java/com/amazon/opendistroforelasticsearch/security/support/ConfigConstants.java

```diff
@@ -236,6 +237,7 @@
     public static final String OPENDISTRO_SECURITY_RESTAPI_ENDPOINTS_DISABLED = "opendistro_security.restapi.endpoints_disabled";
     public static final String OPENDISTRO_SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX = "opendistro_security.restapi.password_validation_regex";
     public static final String OPENDISTRO_SECURITY_RESTAPI_PASSWORD_VALIDATION_ERROR_MESSAGE = "opendistro_security.restapi.password_validation_error_message";
+    public static final Pattern OPENDISTRO_SECURITY_RESTAPI_RESOURCE_NAME_REGEX = Pattern.compile("^[a-zA-Z0-9_-]+$"); // alphanumeric characters, underscores and hyphens
```

Comment:
This restriction is too strong and it will prevent resources with "." or any non ASCII letters.

Reply:
Why is it desirable to create resources with non ASCII characters for well defined security configurations like internalusers, roles, action groups ... ? 

Even characters like "." needs to be URL encoded which makes it less readable anyways.

Reply:
User names must support all languages (not only English) and characters like . and @ are quite common in user names as well.

Reply:
Fair point. Will consult for recommendation and close the issue for now